src/arena.rs

```diff
@@ -575,3 +576,94 @@ where
         })
     }
 }
+
+/// A set of arena handles, represented as a bit vector.
+///
+/// This represents a set of `Handle`s simply by storing a bit for every
+/// `Handle` index. The underlying bit vector is sized dynamically, so storage
+/// depends on the highest handle index that has been inserted.
+///
+/// A `HandleBitSet<T>` will only hold `Handle<T>` values, so it's a bit more
+/// strictly typed than using `BitSet` directly.
+///
+/// This is a good representation for a handle set if you expect it to typically
+/// contain many of the arena's handles. In that situation, `HandleBitSet` will use
+/// a fraction of the memory of a `HashSet`. But if you expect the set to hold only
+/// a small proportion of the arena's handles, `HashSet` would be a better choice.
+pub struct HandleBitSet<T> {
```

Comment:
if we are to have it, perhaps we can move it somewhere closer to the place where it's needed (i.e. a submodule `valid/handle_set.rs`).